cmd/object-handlers.go

```diff
@@ -1758,8 +1758,10 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		md5hex = "" // Do not try to verify the content.
 		sha256hex = ""
 	}
-
-	hashReader, err := hash.NewReader(reader, size, md5hex, sha256hex, actualSize)
+	if crypto.S3KMS.IsRequested(r.Header) {
```

Comment:
We should also make sure `content-md5` is not requested, if content-md5 is requested we must ensure that we calculate `md5sum` as per protocol regardless of SSE-KMS is specified or not.